From: Jiri Slaby <jslaby@suse.cz>
Subject: x86/spectre_v2: nospectre_v2 means nospec too
Patch-mainline: Never, SUSE specific
References: bsc#1075994 bsc#1075091 bnc#1085958

In patch patches.suse/21-x86-spec-add-nospec-chicken-bit.patch, we add
"nospec" kernel parameter to disable spectre. Retpolines in 4.4.113 add
"nospectre_v2". Make sure the latter disables also the former.

Note that we have to parse this as eaerly params, it's too late in
spectre_v2_parse_cmdline, for details see bnc#1085958.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/x86/kernel/cpu/spec_ctrl.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- a/arch/x86/kernel/cpu/spec_ctrl.c
+++ b/arch/x86/kernel/cpu/spec_ctrl.c
@@ -86,3 +86,19 @@ static int __init nospec(char *str)
 	return 0;
 }
 early_param("nospec", nospec);
+
+static int __init nospectre_v2(char *str)
+{
+	return nospec(str);
+}
+early_param("nospectre_v2", nospectre_v2);
+
+static int __init spectre_v2(char *str)
+{
+	if (str && !strcmp(str, "off"))
+		return nospec(str);
+
+	return 0;
+}
+early_param("spectre_v2", spectre_v2);
+
